### ISSUE TITLE
Last message duplicated switching to a new channel after upgrade (wtih separate date header the first time viewed)

### ISSUE THREAD

**Issue by GreenReaper** [Nheko-Reborn/nheko#392]
### Describe the bug
After upgrading via installer to Nheko 0.80 from the last release, the first channel in our third-party server loaded correctly.

I switched to the second channel, and the last message was already visible (it had been loaded to show it in the sidebar already). However, when the rest of them loaded automatically, the last message was duplicated above the existing instance, with two instances of the last message's date header and message, one after the other.

When I switched back to the first channel, and then back to the second, the duplicate date header had been collapsed, but  there were still duplicate messages. Both were the same raw message. This issue persists after exiting and restarting the application.

### To Reproduce
1. Install the new version with the new database code (on top of an existing database?)
2. Click on a channel (or private chat) which already has multiple messages in, one of which has already been loaded
3. Wait for it to load the rest.
4. See initial error.
5. Switch back to the first channel
6. Switch back to the second channel.
7. See that the date header has been collapsed, but there are still two messages

### Expected behavior
The last message should not appear duplicated, either the first time or subsequent times the channel is viewed. Any other messages on that date should be loaded above it inside the same, single date header for that date.

### Screenshots
Example of first view after initial loading of an individual channel after upgrade:
![InitialDuplicateMessage](https://user-images.githubusercontent.com/383043/105217338-a262f080-5b4b-11eb-88fb-09c22abaa663.png)
Example of all subsequent views (a different chat):
![DuplicateMessage](https://user-images.githubusercontent.com/383043/105217335-a1ca5a00-5b4b-11eb-9032-504a94fccf74.png)

### System:

- Nheko version: 0.8.0-a9faecea | Windows
- Installation method: Release installer
- Operating System: Windows 10 Home 20H2 build 19042.746

### Logs
```
[2021-01-20 17:34:16.337] [qml] [warning] QVariant::load: unknown user type with name UserSettings::Presence. (:0, )
[2021-01-20 17:34:21.692] [db] [info] Running migrations, this may take a while!
[2021-01-20 17:34:21.832] [db] [info] Successfully deleted pending receipts database.
[2021-01-20 17:34:21.833] [db] [info] Successfully migrated olm sessions.
[2021-01-20 17:34:21.833] [db] [info] Migrations finished.
[2021-01-20 17:34:21.833] [db] [info] restoring state from cache
[2021-01-20 17:34:21.841] [ui] [info] initialize room list
[2021-01-20 17:34:21.857] [ui] [info] Trying to activate room ![room 1]
[2021-01-20 17:34:22.023] [ui] [info] Activated room ![room 1]
[2021-01-20 17:34:22.035] [crypto] [info] ed25519   : [ed]
[2021-01-20 17:34:22.035] [crypto] [info] curve25519: [curve]
[2021-01-20 17:34:22.038] [ui] [info] jdenticon plugin not found.
[2021-01-20 17:34:22.067] [ui] [info] timelineWidth: -242
[2021-01-20 17:34:22.067] [ui] [info] timelineWidth: 18446744073709551374, min 630
[2021-01-20 17:34:23.513] [ui] [info] timelineWidth: 1338
[2021-01-20 17:34:23.513] [ui] [info] timelineWidth: 1338, min 630
[2021-01-20 17:34:24.261] [ui] [info] starting nheko 0.8.0-a9faecea
[2021-01-20 17:34:24.309] [qml] [warning] libpng warning: iCCP: known incorrect sRGB profile (:0, )
[2021-01-20 17:34:25.255] [qml] [warning] QObject::killTimer: Timers cannot be stopped from another thread (:0, )
[2021-01-20 17:34:25.256] [qml] [warning] QObject::startTimer: Timers cannot be started from another thread (:0, )
[2021-01-20 17:34:30.752] [ui] [info] timelineWidth: 1338
[2021-01-20 17:34:30.752] [ui] [info] timelineWidth: 1338, min 630
[2021-01-20 17:34:32.073] [ui] [info] Trying to activate room ![room 2]
[2021-01-20 17:34:32.145] [qml] [warning] libpng warning: iCCP: known incorrect sRGB profile (:0, )
[2021-01-20 17:34:32.158] [ui] [info] Activated room ![room 2]
[2021-01-20 17:34:37.409] [ui] [info] Trying to activate room ![room 1]
[2021-01-20 17:34:38.487] [ui] [info] Activated room ![room 1]
```

The log file was (apparently incorrectly, according to instructions) located in `C:/Users/<USER>/AppData/Local/nheko/nheko/cache`

**Comment by deepbluev7** [comment]
Does that only apply to the first messages after the upgrade?

**Comment by GreenReaper** [comment]
It applies to the last message before the upgrade (which is why I mentioned the sidebar that shows those, in case it is relevant).

It does not appear to cause further duplicates if I post a new message in the relevant channels, exit, and restart; however, existing duplicated messages remain visible as in the second screenshot.
________________________________

**Comment by deepbluev7** [comment]
Yeah, that is a complication with the database upgrade. Some information is lost, which may cause it to duplicate a few messages on upgrade. You can fix that after a new message was received in that room by running /clear-timeline. I'll close this, since this is not really something we will fix and since it only applies to the 0.8.0 upgrade.
